main/static/script.js

```diff
@@ -1,13 +1,16 @@
 "use strict";
 
-console.log("Hello, world!");
-
 var module = angular.module("main",[]);
 
+var chartsReady = false;
+
+
+
 module.controller("mainCtrl", function($scope, $http) {
 	$scope.queryResult = "";
 	$scope.queryText = "";
 	$scope.onSubmitQuery = function() {
+		window.location.hash = "#" + encodeURIComponent($scope.queryText);
```

Comment:
oh, you should use `$location` and `$path` - those provide a bit better abstraction over the current locations.